libraries/lib-music-information-retrieval/MirDsp.cpp

```diff
@@ -0,0 +1,138 @@
+#include "MirDsp.h"
+#include "FFT.h"
+#include "MirAudioReader.h"
+#include "MirTypes.h"
+#include "MirUtils.h"
+#include "StftFrameProvider.h"
+
+#include <cassert>
+#include <cmath>
+#include <numeric>
+
+namespace MIR
+{
+namespace
+{
+float GetNoveltyMeasure(
+   const std::vector<float>& prevPowSpec, const std::vector<float>& powSpec)
+{
+   auto k = 0;
+   return std::accumulate(
+      powSpec.begin(), powSpec.end(), 0., [&](float a, float mag) {
+         // Half-wave-rectified stuff
+         return a + std::max(0.f, mag - prevPowSpec[k++]);
+      });
+}
+
+std::vector<float>
+GetMovingAverage(const std::vector<float>& x, double sampleRate)
```

Comment:
The argument is called `sampleRate` but at the point of usage it is passed a frame rate, which is sample rate divided by a hop size.